pyamplipi/docs/cli.md

```diff
@@ -0,0 +1,184 @@
+# Commane Line Interface
```

Comment:
We probably will want to link to this from the main README

Reply:
> We've been playing around with the interface for a couple of hours and everything is very usable.
> 
> I found a couple of small issues in shell mode that kicked me out:
> 
> Entering the topic correctly but forgetting the action:
> 

yes, and this moves the argparse to go into `--help` mode which in turn does a system exit
I tried resolving that through `exit_on_error = False`  but only helped for parse-errors that don´t trigger the print_help()

haven´t found another solution yet

> 
> Also when running this through my standard typechecker setup of pylint + mypy I found and fixed some type issues I added them to the PR in the hopes that they are helpful. Feel free to remove them.

interesting.

if that typechecking is available as a standalone python package than we should look into  (1) adding it to the `requirements-dev.txt` and (2) include in the `make check` 


I have no intention at all to remove your additions / modifications -- will investigate later, but I am all ok with you proceeding with the merge...